packages/swirl-components/src/components/swirl-form-control/swirl-form-control.tsx

```diff
@@ -173,7 +173,7 @@ export class SwirlFormControl {
   private onKeyDown = (event: KeyboardEvent) => {
     if (event.key === "Tab") {
       setTimeout(() => {
-        if (!this.el.contains(getActiveElement())) {
+        if (!isDescendantOf(getActiveElement(), this.el)) {
```

Comment:
The issue with the tab navigation was due to this.
As the popover was inside a Shadow Root, it wasn't considered a descendent of the control itself, so the input would lose focus here.
I then implemented the `isDescendantOf ` utility method which checks whether an element is a descendant of another, even if it is inside a Shadow Root.